src/Executable.js

```diff
@@ -279,7 +342,17 @@ export default class Executable {
 
             switch (this._shouldRetry(request, response)) {
                 case ExecutionState.Retry:
-                    await delayForAttempt(attempt);
+                    if (this._maxBackoff == null) {
```

Comment:
This conditional should be moved to the beginning of this method since it essentially no-ops after the first iteration.

Reply:
I meant the beginning of the `execute` method. The point is that by having the conditions within the switch statement they'll be executed each time the request is retried which doesn't make much sense. It'd be better to simply execute them ones ahead of time. You're not wrong that by having them within the switch statement they might never be executed making it even more efficient, but the likeliness of that happening is quite slim; speaking from experience.